src/components/Tasks/AddNewTaskItem.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useContext } from 'react';
+
+import ICAL from 'ical.js';
+import uuid from 'uuid';
+
+import Checkbox from '@material-ui/core/Checkbox';
+import TextField from '@material-ui/core/TextField';
+
+import { TaskType, TaskStatusType, TaskPriorityType } from '../../pim-types';
+import { ListItem } from '../../widgets/List';
+import { PimContext } from '../../Pim';
+
+const AddNewTaskItem = () => {
+  const [title, setTitle] = useState('');
+  const { onItemSave: save, collectionsTaskList: collections } = useContext(PimContext);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+  };
+
+  const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+
+      const event = new TaskType(null);
+      event.uid = uuid.v4();
+      event.title = title;
+      event.status = TaskStatusType.NeedsAction;
+      event.priority = TaskPriorityType.None;
```

Comment:
Is setting status+priority needed? Aren't these the default? I really have no idea and just asking.

Reply:
I believe priority is automatically set (to 0), but status is not. Should I set the status in the constructor?

Reply:
Sorry, let me clarify what I meant: you don't have to set all the fields. Specifically, don't set fields that haven't been changed from the default. The reason for that is that if you don't set fields, they don't show in the resulting ical, which is a good thing. No need to add noise there.